nucleus/experimental/hosted_inference_client.py

```diff
@@ -0,0 +1,262 @@
+        raise NotImplementedError
+
+    def get_model_endpoints(self) -> List[ModelEndpoint]:
+        """
+        Gets all model endpoints that the user owns.
+        """
+        resp = self.connection.get("endpoints")
+        return [
+            ModelEndpoint(endpoint_id=endpoint_id, client=self)
+            for endpoint_id in resp
+        ]
+
+    def edit_model_endpoint(self):
+        # TODO args, corresponds to PUT model_endpoint, doesn't exist serverside
+        raise NotImplementedError
+
+    def sync_request(self, endpoint_id: str, s3url: str):
+        """
+        Makes a request to the Model Endpoint at endpoint_id, and blocks until request completion or timeout.
+
+        Parameters:
+            endpoint_id: The id of the endpoint to make the request to
+            s3url: A url that points to a file containing model input.
+                Must be accessible by Hosted Inference, hence it needs to either be public or a signedURL.
+
+        Returns:
+            A signedUrl that contains a cloudpickled Python object, the result of running inference on the model input
+        """
+        resp = self.connection.post(
+            payload=dict(url=s3url), route=f"task/{endpoint_id}"
+        )
+        return resp["data"]["result_url"]
+
+    def async_request(self, endpoint_id: str, s3url: str):
+        """
+        Makes a request to the Model Endpoint at endpoint_id, and immediately returns a key that can be used to retrieve
+        the result of inference at a later time.
+
+        Parameters:
+            endpoint_id: The id of the endpoint to make the request to
+            s3url: A url that points to a file containing model input.
+                Must be accessible by Hosted Inference, hence it needs to either be public or a signedURL.
+
+        Returns:
+            An id/key that can be used to fetch inference results at a later time
+        """
+        resp = self.connection.post(
+            payload=dict(url=s3url), route=f"task_async/{endpoint_id}"
+        )
+        return resp["data"]["task_id"]
+
+    def get_async_response(self, async_task_id: str):
+        """
+        Gets inference results from a previously created task.
+
+        Parameters:
+            async_task_id: The id/key returned from a previous invocation of async_request.
+
+        Returns:
+            A dictionary that contains task status and optionally a result url if the task has completed.
+            Dictionary's keys are as follows:
+            state: 'PENDING' or 'SUCCESS' or 'FAILURE'
+            result_url: a url pointing to inference results. This url is accessible for 12 hours after the request has been made.
+        TODO: do we want to read the results from here as well? i.e. translate result_url into a python object
+        """
+
+        resp = self.connection.get(route=f"task/result/{async_task_id}")
+        return resp["data"]
+
+    def batch_request(self, endpoint_id, s3urls: List[str]):
```

Comment:
I made a comment on @Tianwei-She 's PR here: https://github.com/scaleapi/models/pull/2315#discussion_r784440984 - wondering if we can just get rid of this method.